emission/tests/funcTests/TestFunctionTiming.py

```diff
@@ -0,0 +1,151 @@
+# emission/tests/funcTests/TestFunctionTiming.py
+
+import logging
+import time
+import typing as t
+import pymongo
+
+# Import the store_dashboard_time and store_dashboard_error functions
+import emission.storage.decorations.stats_queries as sdq
+
+# Import the existing Timer context manager
+import emission.core.timer as ec_timer
+
+# Import the database module for verification
+import emission.storage.timeseries.abstract_timeseries as esta
+
+# Define test functions
+def test_function_1():
+    logging.info("Executing test_function_1")
+    time.sleep(1)  # Simulate processing time
+    return True  # Indicate successful execution
+
+def test_function_2():
+    logging.info("Executing test_function_2")
+    time.sleep(2)
+    return True
+
+def test_function_faulty():
+    logging.info("Executing test_function_faulty")
+    time.sleep(1)
+    raise ValueError("Simulated error in test_function_faulty")
+
+def test_function_3():
+    logging.info("Executing test_function_3")
+    time.sleep(3)
+    return True
+
+def execute_and_time_function(func: t.Callable[[], bool]):
+    """
+    Executes a given function, measures its execution time using ECT_Timer,
+    stores the timing information using store_dashboard_time, and verifies
+    that the data was stored successfully by querying the timeseries database.
+    If the function raises an exception, it stores the error using store_dashboard_error
+    and verifies the error storage.
+
+    Parameters:
+    - func (Callable[[], bool]): The test function to execute and time.
+    """
+    function_name = func.__name__
+    timestamp = time.time()
+
+    logging.info(f"Starting timing for function: {function_name}")
+
+    try:
+        with ec_timer.Timer() as timer:
+            result = func()  # Execute the test function
+
+        elapsed_seconds = timer.elapsed  # Accessing the float attribute directly
+        elapsed_ms = elapsed_seconds * 1000  # Convert to milliseconds
+
+        # Store the execution time
+        sdq.store_dashboard_time(
+            code_fragment_name=function_name,
+            ts=timestamp,
+            reading=elapsed_ms
+        )
```

Comment:
Why not have `store_dashboard_time` accept `timer` as a parameter? That way, you won't need to repeat these lines of code everywhere you want to instrument

Reply:
Then to verify that the reading was stored, check the most recent `stats/dashboard_time` entry and validate the `reading` lines up with `timer.elapsed`

(To check the most recent entry, should we implement `find_one` on the aggregate timeseries?)

Reply:
> Why not have `store_dashboard_time` accept `timer` as a parameter? 

That is a great suggestion, @JGreenlee, and a clear refinement of my previous implementation. I think that what @TeachMeTW has implemented now is good enough to unblock us, but maybe we can go back to this at some time in the future and refine further (convert the pipeline time functions to this form as well, put the stat name string into the timer etc)